platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/oauth/OAuthSecurityImpl.java

```diff
@@ -113,6 +108,68 @@ public void setUserTokenOnClient(
     client.header(OAUTH, BEARER + tokenEntry.getAccessToken());
   }
 
+  /**
+   * Gets the user's access token from the token storage to set it to the OAUTH header. If one is
+   * not available, make a call to the OAuth provider to get tokens.
+   *
+   * @param client Non-null client to set the access token on.
+   * @param sourceId the id of the source using OAuth needed to get the correct tokens
+   * @param clientId The client ID registered with the OAuth provider
+   * @param clientSecret The client secret registered with the OAuth provider
+   * @param username The user's username
+   * @param password The user's password
+   * @param discoveryUrl the metadata URL of the OAuth provider
+   */
+  public void setUserTokenOnClient(
+      Client client,
+      String sourceId,
+      String clientId,
+      String clientSecret,
+      String username,
+      String password,
+      String discoveryUrl,
+      Map<String, String> additionalParameters) {
+
+    if (client == null || Strings.isBlank(sourceId) || Strings.isBlank(username)) {
+      return;
+    }
+
+    TokenInformation.TokenEntry tokenEntry = tokenStorage.read(username, sourceId);
+    if (tokenEntry != null
+        && discoveryUrl.equalsIgnoreCase(tokenEntry.getDiscoveryUrl())
+        && !isExpired(tokenEntry.getAccessToken())) {
```

Comment:
❓ Should this expiry check also be added to the above method after line 102?

Reply:
No like I mentioned above this is the second half of the `OAuthPlugin` which takes care of all of that